### ISSUE TITLE
Cannot set user secret starting with - in the CLI

### ISSUE THREAD

**Issue by Bartmax** [dotnet/aspnetcore#21283]
running: 
```
dotnet user-secrets set "hi" "-bart"
```
error:
```
Specify --help for a list of available options and commands.
Unrecognized option '-bart'
```

I found no way to set a secret when it's start with the `-` key. I had to update the .json file manually.

I would expected that if the value is within quotes " it should have worked.

<details>
<summary>Click to see dotnet --info output :</summary>
<blockquote>
.NET Core SDK (reflecting any global.json):
 Version:   3.1.300-preview-015095
 Commit:    85c45057dc

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.300-preview-015095\

Host (useful for support):
  Version: 3.1.3
  Commit:  4a9f85e9f8

.NET Core SDKs installed:
  1.1.14 [C:\Program Files\dotnet\sdk]
  3.1.102 [C:\Program Files\dotnet\sdk]
  3.1.200-preview-014977 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015095 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
</blockquote>
</details>

**Comment by javiercn** [comment]
@Bartmax thanks for contacting us.

The OS takes out the quotes I believe so dotnet user-secrets is likely thinking that's a parameter. I don't know if there's a way to escape that sequence.

**Comment by Bartmax** [comment]
tried `\` and `@` with no luck.

**Comment by javiercn** [comment]
@anurse or someone on his team might know. Otherwise, I would check the source code for the tool, it should be relatively straightforward to figure out from there if there's a way.

**Comment by anurse** [comment]
Pretty sure this is a known bug and has been fixed in 5.0 (though I don’t have the bug number handy)

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @Bartmax.
Closing as per @anurse 's comment this is now fixed in the latest 5.0 preview release. If this is not the case, please file a new issue and reference this one.
